test/controllers/trips_controller_test.rb

```diff
@@ -1,23 +1,99 @@
 require "test_helper"
 
 describe TripsController do
+  let (:trip) {
+    Trip.create
+  }
+
+  describe "index" do 
+    it "can get a list of all trips" do 
+
+      get trips_path 
+
+      must_respond_with :success 
+    end 
+  end 
+
   describe "show" do
-    # Your tests go here
+
+    it "can get trip" do 
+      passenger = Passenger.create(name: "Nina", phone_num: "12345")
+      driver = Driver.create(name: "Anna", vin: "qwert123", available: true)
+      new_trip = Trip.create(date: "today", rating: 0, cost: Trip.trip_cost, driver_id: driver.id, passenger_id: passenger.id)
+
+     
+      get trip_path(new_trip.id)
+      must_respond_with :success
+    end 
+
+    it "will redirect for invalid trip" do 
+      get trip_path(-1)
+      must_respond_with :redirect
+
+    end 
   end
 
   describe "create" do
-    # Your tests go here
+    it "can request new trip" do 
+
+      passenger = Passenger.create(name: "Nina", phone_num: "12345")
+      driver = Driver.create(name: "Anna", vin: "qwert123", available: true)
+
+      trip_hash = {
+        trip: {
+          date: "today",
+          rating: 0,
+          cost: Trip.trip_cost,
+          driver_id: driver.id,
+          passenger_id: passenger.id
+        }
+      }
+
+      expect {
+        post passenger_trips_path(passenger.id), params: trip_hash
+      }.must_differ "Trip.count", 1
+
+      must_redirect_to passenger_path(passenger.id)
+
+    end 
   end
 
   describe "edit" do
-    # Your tests go here
+    it "responds with success when getting to the edit trip's rating page" do 
+
+      passenger = Passenger.create(name: "Nina", phone_num: "12345")
+      driver = Driver.create(name: "Anna", vin: "qwert123", available: true)
+      new_trip = Trip.create(date: "today", rating: 0, cost: Trip.trip_cost, driver_id: driver.id, passenger_id: passenger.id)
```

Comment:
I know `Trip.trip_cost` is essentially a random number generator, but within the tests, it may be more useful to just hardcode a random number. (For example, what if you're debugging this test, but it turns out that `Trip.trip_cost` was the broken method all along?-- you want to reduce even the dependency that this test has on a working `Trip.trip_cost` implementation)